lib/taskrunner.js

```diff
@@ -33,6 +33,26 @@ module.exports = {
       params.overrides.containerOverrides[0].environment = options.env;
     }
 
+    if (options.fargate !== undefined) {
+      params.launchType = 'FARGATE';
+    }
+
+    if (options.subnets !== undefined || options.securityGroups !== undefined) {
+      params.networkConfiguration = {
+        awsvpcConfiguration: {
+          assignPublicIp: 'DISABLED'
+        }
+      };
+    }
+
+    if (options.subnets !== undefined) {
```

Comment:
both of these options should really only be allowed if awsvpc networking is enabled, or at least if Fargate is enabled.

Reply:
Validation is added in index.js. Since the networkMode is defined on the taks definition instead of the runtask request it makes sense to validate there.